refactor to use `util/bklog.G(ctx)` instead `logrus.` directly

morlay opened the pull request:
Too may changes

I'm not sure, Should I add new param `ctx context.Context`, specially for public functions

tonistiigi commented on util/bklog/log.go:
```diff
@@ -0,0 +1,46 @@
+
+// WithLogger returns a new context with the provided logger. Use in
+// combination with logger.WithField(s) for great effect.
+func WithLogger(ctx context.Context, logger *logrus.Entry) context.Context {
```
Don't see this used anywhere. Maybe not add it until there is a need.

morlay replied:
Used here, may with section id for logs. 
so we could know which logs belongs to sections 

https://github.com/moby/buildkit/pull/2236/files#diff-527db53529b974ea055e9d1d51a63c3c0b7dbf8cc61b03eb3d9693604333a437R112

tonistiigi commented on control/control.go:
```diff
@@ -451,23 +453,23 @@ func (c *Controller) gc() {
 }
 
-func parseCacheExportMode(mode string) solver.CacheExportMode {
+func parseCacheExportMode(ctx context.Context, mode string) solver.CacheExportMode {
```
another way to fix this is to make it return `(solver.CacheExportMode, bool)`

tonistiigi reviewed (commented):
Generally looks ok to me. Do I understand correctly that `bklog.G(ctx)` can be stored as a value and use multiple times to generate a log?

morlay commented:
> bklog.G(ctx) can be stored as a value and use multiple times to generate a log?

Yes, but could only as variable in a function scope

morlay commented:
@tonistiigi 

* removed all `logrus.` in each pkg (exclude test files).
* added context.Context as  first parameter if required.
* use `bklog.G(context.TODO())` for hard to refactor, like `solver`, may next PR to do more refactor.

tonistiigi commented on client/solve.go:
```diff
@@ -109,7 +109,9 @@ func (c *Client) solve(ctx context.Context, def *llb.Definition, runGateway runG
-	cacheOpt, err := parseCacheOptions(opt)
+	statusContext = bklog.WithLogger(statusContext, bklog.GetLogger(statusContext).WithField("session", s.ID()))
+
+	cacheOpt, err := parseCacheOptions(statusContext, opt)
```
why pass statusContext in here instead of `ctx`. parsing doesn't seem to be related to session

morlay replied:
I think they are in same scope `resolve`, should with same logger and same tracing.

tonistiigi replied:
I don't understand. In fact, I think it would be more correct if this `parse` func was called before the session is even created.

morlay replied:
I just think in tracing view.

The flow in `solve`.

```
Solve:

NewSection(ctx) ->  parseCache(ctx) -> doOthers(ctx) 
```

I feel the ctx should be the same one.

morlay replied:
I revert this.
this change may out of the PR scope.

tonistiigi commented on worker/base/worker.go:
```diff
@@ -90,6 +90,10 @@ type Worker struct {
+	return NewWorkerWithContext(context.TODO(), opt)
+}
+
+func NewWorkerWithContext(ctx context.Context, opt WorkerOpt) (*Worker, error) {
```
Just change the signature

morlay replied:
may break other pkgs which use `buildkit`.

tonistiigi replied:
Then they update. We don't care about go API stability for these packages where it adds maintenance cost.
